org.eclipse.xtext.util/src/org/eclipse/xtext/util/IFileSystemScanner.xtend

```diff
@@ -29,13 +29,19 @@ interface IFileSystemScanner {
         }
         
         def void scanRec(File file, IAcceptor<URI> acceptor) {
-            acceptor.accept(URI.createFileURI(file.absolutePath))
+            val uri = URI.createURI(file.absoluteFile.toURI.toPath)
+            acceptor.accept(uri)
             if (file.isDirectory) {
                 for (f : file.listFiles) {
                     scanRec(f, acceptor)
                 }
             }
         }
         
+        def String toPath(java.net.URI uri) {
```

Comment:
the name 'toPath' is missleading if it returns a string representation of a URI.